render/gles2/pixel_format.c

```diff
@@ -39,10 +39,41 @@ static const struct wlr_gles2_pixel_format formats[] = {
 		.gl_type = GL_UNSIGNED_BYTE,
 		.has_alpha = true,
 	},
+		// 10 bit formats
```

Comment:
I don't think this is enough to support these buffer formats. I'd assume we need shaders as well?

Have you tested a 10-bit client?

Reply:
Also, this list is only used for `wl_shm` clients. GPU-accelerated clients don't use it.

Reply:
I have a 10-bit branch of sway I tested with my 10-bit wlroots branch some time ago. I used the method of running my compiled version of sway/wlroots like in the wiki, and start it with a line in my `~/.zprofile` so that my compiled version of sway gets used on login instead of the package managed version. It worked mostly as expected, aside from the mouse cursor having a corrupted sprite/texture. However, I'm not sure if it was from my changes, or if it was from some other commits/changes. At the time it was mostly just applying @tadeokondrak's patch, and compiling that with sway. I started some very initial work with 10-bit color in sway that includes the aforementioned output argument. I'm not sure if the output argument was working properly, since I think the patch I started with may have been hard coding the `gbm_format` stuff. I was able to log in.

It's been awhile since I've tested it that way. I wasn't confident with the HDMI cable I was using then, so since then I have gotten a proper certified 2.0b cable. I've been using a HDR capable smart tv. I can toggle between YCbCr and RGB in the TV settings. Assuming the TV's menu is reporting things accurately, it should also tell me if it is receiving 10-bit HDR input. I'll need to check if just having it be 10-bit color is enough for that, or if 10-bit HDR is a more specific specification than that. I'm assuming that it may need to be YCbCr for it to work, but the RGB option on the TV makes me think it might be more flexible. If not, I'll need to check my collection of monitors to see if any of them are 10-bit or not.

> Also, this list is only used for `wl_shm` clients. GPU-accelerated clients don't use it.

Duly noted. I'll be looking into that along with the shaders you've mentioned.

Reply:
I've tested my more recent commits. The sway log is spitting out some errors.

```
00:01:16.638 [ERROR] [EGL] command: eglCreatePlatformWindowSurfaceEXT, error: EGL_BAD_MATCH (0x3009), message: "EGL config not compatible with GBM format"
00:01:16.638 [ERROR] [render/egl.c:406] Failed to create EGL surface
00:01:16.638 [ERROR] [backend/drm/renderer.c:103] Failed to create EGL surface
00:01:16.638 [ERROR] [backend/drm/drm.c:895] Cannot allocate cursor resources
00:01:16.638 [DEBUG] [types/wlr_output.c:1050] Falling back to software cursor on output 'HDMI-A-4'
```

I can log in, but the TV is telling me it is not getting HDR input with either the RGB or YCbCr input setting on the TV. AFAIK, the EGL stuff has to do with shaders, so I'll need to look closer at what is going on with the GBM format. Given there was a fallback to software cursors and the older issue I had was a garbled cursor, I'm wondering if I had a similar EGL/GBM issue with the older issue as well.


I'll test again after adding something for GPU-accelerated clients. It looks like my login session was GPU-accelerated. It seems I didn't add color formats for GPU-accelerated clients, so mine couldn't find what it needed for 10-bit color.

Reply:
>This is surprising to me. Isn't the GPU responsible for switching between YCbCr and RGB?

As far as I'm aware. The colors do visibly change when I switch between YCbCr and RGB. It's a Vizio if you're curious. It looks like some Samsung TVs may also have this option. I guess it just asks the source for YCbCr/RGB output? As I look into it more, it seems to be for PC input. I guess some GPUs might not handle YCbCr for some reason or another? Or maybe it's a weird manufacturer design decision?